conda_devenv/devenv.py

```diff
@@ -0,0 +1,248 @@
+    return merged_dict, environment
+
+
+def render_for_conda_env(yaml_dict):
+    import yaml
+    return yaml.dump(yaml_dict, default_flow_style=False)
+
+
+def render_activate_script(environment):
+    script = []
+    if sys.platform.startswith("linux"):
+        script = ["#!/bin/sh"]
+    for variable, value in environment.items():
+        if sys.platform.startswith("linux"):
+            if isinstance(value, list):
+                # Lists are supposed to prepend to the existing value
+                value = os.pathsep.join(value) + os.pathsep + "${variable}".format(variable=variable)
+
+            script.append("export CONDA_DEVENV_BKP_{variable}=${variable}".format(variable=variable))
+            script.append("export {variable}=\"{value}\"".format(variable=variable, value=value))
+
+        elif sys.platform.startswith("win"):
+            if isinstance(value, list):
+                # Lists are supposed to prepend to the existing value
+                value = os.pathsep.join(value) + os.pathsep + "%{variable}%".format(variable=variable)
+
+            script.append("set CONDA_DEVENV_BKP_{variable}=%{variable}%".format(variable=variable))
+            script.append("set {variable}=\"{value}\"".format(variable=variable, value=value))
+
+        else:
+            raise ValueError("Unknown platform")
+
+    return '\n'.join(script)
+
+
+def render_deactivate_script(environment):
+    script = []
+    if sys.platform.startswith("linux"):
+        script = ["#!/bin/sh"]
+    for variable in environment.keys():
+        if sys.platform.startswith("linux"):
+            script.append("export {variable}=$CONDA_DEVENV_BKP_{variable}".format(variable=variable))
+            script.append("unset CONDA_DEVENV_BKP_{variable}".format(variable=variable))
+
+        elif sys.platform.startswith("win"):
+            script.append("set {variable}=%CONDA_DEVENV_BKP_{variable}%".format(variable=variable))
+            script.append("set CONDA_DEVENV_BKP_{variable}=".format(variable=variable))
+
+        else:
+            raise ValueError("Unknown platform")
+
+    return '\n'.join(script)
+
+
+def __write_conda_environment_file(args, filename, rendered_contents):
+    if args.output_file:
+        output_filename = args.output_file
+    else:
+        output_filename = filename.rstrip(".devenv.yml") + ".yml"
+        if output_filename.endswith(".yml.yml"):
+            raise ValueError("Can't guess the output file, please provide the output file with the --output-filename "
+                             "flag")
+
+    if os.path.exists(output_filename) and not args.force:
+        raise ValueError("The output file '{}' already exists and the flag --force was omitted. Refusing to override "
+                         "the file.".format(output_filename))
+
+    with open(output_filename, 'w') as f:
+        f.write(rendered_contents)
+
+    return output_filename
+
+
+def __call_conda_env_update(args, output_filename):
```

Comment:
That's a good question. The ones that have the `__` prefix don't have specific unit tests.

What do you guys think about this?